src/ring_theory/polynomial/pochhammer.lean

```diff
@@ -92,6 +92,26 @@ begin
     refl, },
 end
 
+lemma pochhammer_add_pochhammer_aux {R : Type*} [comm_semiring R] : ∀ (n : ℕ),
+  pochhammer R n + n * (pochhammer R (n - 1)).comp (X + 1) = (pochhammer R n).comp (X + 1)
```

Comment:
Is it worth avoiding subtraction here?
```suggestion
  (pochhammer R (n + 1)).comp (X + 1) = pochhammer R (n + 1) + (n + 1) * (pochhammer R n).comp (X + 1) = 
```

Reply:
Like in the previous PR, I had this version first, but then I noticed that it also holds for `n = 0`.  I think that in this case I have a slight preference for the lemma with `n - 1`.  I could state the lemma without `- 1` and prove it by applying the other one (using `nat.pred_succ` or `nat.succ_pred`, whichever one is the correct one), if you prefer.

Reply:
I think having both and proving the `- 1` version in terms of the `+1` version is reasonable

Reply:
Ok, I'll do that!

Why do you prefer to prove the version without `- 1` first, and not the other way round?

Reply:
Eric, after changing the `aux` lemma, I realized that maybe this discussion should have been about the general version `pochhammer_add_pochhammer`, rather than the technical lemma `pochhammer_add_pochhammer_aux`?